appkit/src/test/scala/org/ergoplatform/appkit/ErgoAuthSpec.scala

```diff
@@ -0,0 +1,41 @@
+package org.ergoplatform.appkit
+
+import org.scalatest.{Matchers, PropSpec}
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import scorex.util.Random
+import sigmastate.interpreter.HintsBag
+
+import java.nio.charset.StandardCharsets
+
+class ErgoAuthSpec extends PropSpec with Matchers with ScalaCheckDrivenPropertyChecks
+  with AppkitTestingCommon {
+
+  property("ErgoAuth address roundtrip") {
+    // server side
+    val serializedSigmaBoolean = ErgoAuthUtils.serializeSigmaBoolean(address)
+    // message to sign should be something random and not repeating
+    val requestedMessage = addrStr + System.currentTimeMillis().toString
+
+    // transferred to client, and now we are on client side
+    val signedMessage = new String(Random.randomBytes(16)) + requestedMessage +
+      new String(Random.randomBytes(32))
+    val signature = new ColdErgoClient(address.getNetworkType, Parameters.ColdClientMaxBlockCost)
+      .execute { ctx: BlockchainContext =>
+
+        val prover = ctx.newProverBuilder().withMnemonic(mnemonic, SecretString.empty()).build()
+        prover.signMessage(ErgoAuthUtils.deserializeSigmaBoolean(serializedSigmaBoolean),
+          signedMessage.getBytes(StandardCharsets.UTF_8),
+          HintsBag.empty)
+      }
+
+    // transferred to server...
+    ErgoAuthUtils.verifyResponse(address, requestedMessage, signedMessage, signature) shouldBe (true)
```

Comment:
Again, it is better to use SigmaBoolean here instead of Address. SigmaBoolean is very important concept which sits between p2pk and arbitrary Address. It is more general than public key, but less general then Address. 
It is better to make this distinction explicit everywhere in Appkit.
I would even suggest to add SigmaProp class next to Address class and wrap SigmaBoolean into it.
NOTE, there is special.sigma.SigmaProp interface in Sigma interpreter which represent exactly such wrapping (see implementation in sigmastate.eval.CSigmaProp)

Reply:
Done